virtcontainers/persist.go

```diff
@@ -171,6 +172,86 @@ func (s *Sandbox) dumpNetwork(ss *persistapi.SandboxState) {
 	}
 }
 
+func (s *Sandbox) dumpConfig(ss *persistapi.SandboxState) {
+	sconfig := s.config
+	ss.Config = persistapi.SandboxConfig{
+		HypervisorType: string(sconfig.HypervisorType),
+		AgentType:      string(sconfig.AgentType),
+		ProxyType:      string(sconfig.ProxyType),
+		ProxyConfig: persistapi.ProxyConfig{
+			Path:  sconfig.ProxyConfig.Path,
+			Debug: sconfig.ProxyConfig.Debug,
+		},
+		ShimType: string(sconfig.ShimType),
+		NetworkConfig: persistapi.NetworkConfig{
+			NetNSPath:         sconfig.NetworkConfig.NetNSPath,
+			NetNsCreated:      sconfig.NetworkConfig.NetNsCreated,
+			DisableNewNetNs:   sconfig.NetworkConfig.DisableNewNetNs,
+			InterworkingModel: int(sconfig.NetworkConfig.InterworkingModel),
+		},
+
+		ShmSize:             sconfig.ShmSize,
+		SharePidNs:          sconfig.SharePidNs,
+		Stateful:            sconfig.Stateful,
+		SystemdCgroup:       sconfig.SystemdCgroup,
+		SandboxCgroupOnly:   sconfig.SandboxCgroupOnly,
+		DisableGuestSeccomp: sconfig.DisableGuestSeccomp,
+	}
+
+	for _, e := range sconfig.Experimental {
+		ss.Config.Experimental = append(ss.Config.Experimental, e.Name)
+	}
+
+	ss.Config.HypervisorConfig = persistapi.HypervisorConfig{
+		NumVCPUs:              sconfig.HypervisorConfig.NumVCPUs,
+		DefaultMaxVCPUs:       sconfig.HypervisorConfig.DefaultMaxVCPUs,
+		MemorySize:            sconfig.HypervisorConfig.MemorySize,
+		MemSlots:              sconfig.HypervisorConfig.MemSlots,
+		KernelPath:            sconfig.HypervisorConfig.KernelPath,
+		ImagePath:             sconfig.HypervisorConfig.ImagePath,
+		InitrdPath:            sconfig.HypervisorConfig.InitrdPath,
+		SharedFS:              sconfig.HypervisorConfig.SharedFS,
+		VirtioFSDaemon:        sconfig.HypervisorConfig.VirtioFSDaemon,
+		DisableBlockDeviceUse: sconfig.HypervisorConfig.DisableBlockDeviceUse,
+		UseVSock:              sconfig.HypervisorConfig.UseVSock,
+		DisableVhostNet:       sconfig.HypervisorConfig.DisableVhostNet,
+	}
+
+	if sconfig.AgentType == "kata" {
+		var sagent KataAgentConfig
+		err := mapstructure.Decode(sconfig.AgentConfig, &sagent)
+		if err != nil {
+			s.Logger().WithError(err).Error("internal error: KataAgentConfig failed to decode")
+		} else {
+			ss.Config.KataAgentConfig = &persistapi.KataAgentConfig{
+				LongLiveConn: sagent.LongLiveConn,
```

Comment:
Anyone else think of Star Trek when they see this? :smile: :rocket:

```go
spock.liveLongAndProsper(conn)

Reply:
🙌 :raised hand: